CMS-Enterprise/easi-app
# EASI-4672: Deploy to EKS as part of PR pipeline

# EASI-4672

## Description
This pull request includes several changes to the GitHub workflows to improve the build, deployment, and teardown processes for the frontend application. The most important changes include adding new workflows for building the frontend image, deploying to EKS, and tearing down environments on PR close.

### New Workflows:

* Added a new workflow to build the `easi-frontend` Docker image (`.github/workflows/build_frontend_image.yml`).
* Added a new workflow to deploy the branch environment to EKS, which is triggered by a workflow call (`.github/workflows/deploy_to_eks.yml`).
* Added a new workflow for manual deployment to EKS, which includes building the frontend image and application images before deployment (`.github/workflows/manual_deploy_to_eks.yml`).
* Added a new workflow to teardown the branch environment in EKS when a pull request is closed (`.github/workflows/pr_teardown_env_on_close.yml`).

### Modifications to Existing Workflows:

* Updated the main build and test workflow to include steps for building the frontend image and deploying to EKS (`.github/workflows/build_and_test.yml`).

## How to test this change

1. Open new PR
1. Pipeline should deploy env to DEV EKS cluster
1. Once PR closed, EKS env should be deleted

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
